File: Ackermann/Go/jcla1/ackermann.go
package ackermann

func Compute(m, n int) int {
	if m == 0 {
		return n + 1
	} else if m > 0 && n == 0 {
		return Compute(m-1, 1)
	}

	return Compute(m-1, Compute(m, n-1))
}

File: Ackermann/Scala/jcla1/ackermann.scala
def ackermann(m: Int, n: Int): Int = (m, n) match {
    case (0, n) => n + 1
    case (m, 0) => ackermann(m - 1, 1)
    case _ => ackermann(m - 1, ackermann(m, n - 1))
}
